### PR TITLE
Fix case of .yaml and .proto files to all be snake case

### PR DESCRIPTION
**Pull request by KipHamiltons**
There were quite a few errant yaml and proto files. This fixes them.

### CODE REVIEW

File: module/behaviour/skills/HeadBehaviourSoccer/src/HeadBehaviourSoccer.cpp [hunk 1]
```diff
@@ -153,7 +153,7 @@ namespace module::behaviour::skills {
                                                                             [this](const SoccerObjectPriority& p) {
                                                                                 ballPriority = p.ball;
                                                                                 goalPriority = p.goal;
-                                                                                searchType   = p.searchType;
+                                                                                search_type  = p.search_type;
```
Comment (KipHamiltons) [inline module/behaviour/skills/HeadBehaviourSoccer/src/HeadBehaviourSoccer.cpp:156]: this is wrong

File: module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp [hunk 2]
```diff
@@ -80,7 +80,7 @@ namespace module::behaviour::strategy {
         , cfg_()
         , walkTarget()
         , lookTarget()
-        , kickType()
+        , kick_type()
```
Comment (KipHamiltons) [inline module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp:83]: this is also wrong

File: shared/message/motion/WalkCommand.proto [hunk 3]
```diff
@@ -26,22 +26,22 @@ message WalkStarted {}
...
+    vec3   command        = 2;  // x and y are velocity in m/s and angle is in rads/s
 }
 
 message StopCommand {
-    uint64 subsumptionId = 1;  // reservation identifier for servo control
+    uint64 subsumption_id = 1;  // reservation identifier for servo control
 }
 
 message NewWalkCommand {
     vec3 velocityTarget = 1;
```
Comment (CMurtagh-LGTM) [inline shared/message/motion/WalkCommand.proto:38]: :P

**Review by CMurtagh-LGTM — COMMENTED** [review]
Ah yes, the superior case.

**Review by Bidski — DISMISSED** [review]
Sure, LGTM

**Review by Bidski — APPROVED** [review]
LGTM

**Review by Bidski — APPROVED** [review]
OK

**Pull request merged**